engine/gamesys/src/gamesys/scripts/script_buffer.cpp

```diff
@@ -879,6 +880,191 @@ namespace dmGameSystem
         return 0;
     }
 
+
+    // Allocates and fills up an array of ints/floats from a table at the top of the stack. It pops the table before returning.
+    template <typename T>
+    static T* luaTableToArray(lua_State* L, uint32_t count, dmBuffer::ValueType valueType)
+    {
+
+        T* buffer = (T*) malloc(count*dmBuffer::GetSizeForValueType(valueType));
+
+        uint32_t i = 0;
+        lua_pushnil(L);
+        if (valueType == dmBuffer::VALUE_TYPE_FLOAT32) {
+            while (lua_next(L, -2) != 0) {
+                buffer[i] = luaL_checknumber(L, -1);
+                luaL_checknumber(L, -2); // index of the value. Make sure it's a number
+
+                i++;
+                lua_pop(L, 1); // pop value, keep index for next iteration
+            }
+        } else {
+            // array of ints
+            while (lua_next(L, -2) != 0) {
+                buffer[i] = luaL_checkinteger(L, -1);
+                luaL_checknumber(L, -2); // index of the value. Make sure it's a number
+
+                i++;
+                lua_pop(L, 1); // pop value, keep index for next iteration
+            }
+        }
+
+        lua_pop(L, 1); // pop the table itself
+
+        return buffer;
+    }
+
+    // lua: buffer.set_metadata(buf, hash("aabb"), {-1.0, -2.0, -3.0,...}, buffer.VALUE_TYPE_FLOAT32)
+    static int SetMetadata(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        // get buffer
+        dmScript::LuaHBuffer* buffer = dmScript::CheckBuffer(L, 1);
+        dmBuffer::HBuffer hbuffer = UnpackLuaBuffer(buffer);
+
+        // get metadata entry name
+        dmhash_t entry_name = dmScript::CheckHashOrString(L, 2);
+
+        // get number type
+        lua_Integer valueTypeRaw = (dmBuffer::ValueType) luaL_checkinteger(L, 4);
+        dmBuffer::ValueType valueType = (dmBuffer::ValueType) valueTypeRaw;
```

Comment:
Why not `dmBuffer::ValueType valueType = (dmBuffer::ValueType) luaL_checkinteger(L, 4);`?

Reply:
true. Many things learnt on the C/Lua interface while working on this. Many things went under the radar on my final passes on the code. 

Let me fix...